### PR TITLE
omat24 example

### PR DESCRIPTION
**Pull request by allaffa**
@ArCho48 @RylieWeaver 
I would like for you to take a look at how the code is structured, so that you can start getting acquainted with how we import new datasets in HydraGNN.

### CODE REVIEW

**Comment by RylieWeaver** [comment]
@allaffa

Thank you for pointing it out. I will put down my main findings to clarify if I'm understanding things correctly.

Overall, we now support download links and use a FAIRChem class to parse the molecular objects in new datasets.
(1) Download links are defined inside lists and downloaded in a loop based on the task (train/val)
(2) We use the `AseDBDataset`, which gives us functions to easily get traits such as atomic number, position, energy, forces, etc...

File: examples/omat24/train.py [hunk 1]
```diff
@@ -0,0 +1,452 @@
...
+                )
+            )
+
+            rx = list(nsplit(list(range(dataset.num_samples)), self.world_size))[
+                self.rank
+            ]
+
+            for index in iterate_tqdm(rx, verbosity_level=2):
+                try:
+                    xyz = torch.tensor(dataset.get_atoms(index).get_positions())
```
Comment (jychoi-hpc) [inline examples/omat24/train.py:115]: Can you add "dtype=torch.float32" here? 
I got an error: "RuntimeError: mat1 and mat2 must have the same dtype, but got Double and Float"

Comment (allaffa) [reply examples/omat24/train.py:115]:
```suggestion
                    xyz = torch.tensor(dataset.get_atoms(index).get_positions(), dtype=torch.float32)
```

**Review by jychoi-hpc — APPROVED** [review]
I just finished testing. I was able to generate the adios file and run training. The code looks good. Thank you for the update.

**Pull request merged**